File: problems/python/guess-number-higher-or-lower.py
class Solution:
    def guessNumber(self, n: int) -> int:
        l = 1
        r = n + 1
        while r - l > 1:
            pick = (l + r) // 2
            if guess(pick) > -1:
                l = pick
            else:
                r = pick
        return l

File: problems/cpp/guess-number-higher-or-lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return       -1 if num is higher than the picked number
 *                1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        long long l = 1;
        long long r = n + 1LL;
        while(r - l > 1){
            long long m = (l + r) / 2;
            int res = guess(m);
            if(res == 0){
                return m;
            }
            else if(res == 1){
                l = m;
            }
            else{
                r = m;
            }
        }
        return l;
    }
};